explainer.md

```diff
@@ -622,19 +614,24 @@ partial interface Navigator {
   readonly attribute VR vr;
 };
 
-[SecureContext, Exposed=Window] interface VR : EventTarget {
-  attribute EventHandler ondeviceconnect;
-  attribute EventHandler ondevicedisconnect;
+dictionary VRDeviceFilter {
+  boolean exclusive = false;
+};
+
+dictionary VRDeviceRequestOptions {
+  required sequence<VRDeviceFilter> filters;
+};
 
-  Promise<sequence<VRDevice>> getDevices();
+[SecureContext, Exposed=Window] interface VR : EventTarget {
+  attribute EventHandler ondeviceschanged;
```

Comment:
so we don't have the plural vs. singular and active vs. past tenses, do you think `onchange` might read better? (the event names in the event interfaces of the [Web Bluetooth](https://webbluetoothcg.github.io/web-bluetooth/#dom-bluetooth-onavailabilitychanged) and [Sensor](https://www.w3.org/TR/generic-sensor/#the-sensor-interface) APIs are inconsistent, but the [Permissions API uses `onchange`](https://w3c.github.io/permissions/#status-of-a-permission), which I feel is pretty unambiguous. still doesn't disambiguate from `change` vs. `changed`, but it's still better I think.)

Reply:
on second thought, perhaps it's better to be explicit. but, if only one event is emitted per `VRDevice` that is changed, IMO this should be `ondevicechanged` (singular) instead of `ondeviceschanged` (plural).

Reply:
shouldn't there be some `state` associated with the event emitted? from a web developer's POV this is a bit hard to work with:

```js
navigator.vr.addEventListener('devicechange', evt => {
  // Which VR device changed? And to which state did it change?
});
```

see the [Permissions API](https://w3c.github.io/permissions/#status-of-a-permission) for similar cases.

Reply:
I'm not too opinionated on the exact verbiage (though I think "devices" is more accurate than "device", since there may in fact be more than one.)

I don't know what we could appropriately surface in the event without brushing up against the fingerprinting concerns that prompted this change, though. If we deliver the devices in question via this event then developers could presumably just monitor the event to get a more complete view of the system than the query delivers.

The purpose of the event is simply to indicate to developers that if they had previously asked for a device and didn't get an appropriate one back they may now want to try again. Given that, is there an appropriate bit of contextual information that you can think of that would be useful here without describing the available hardware explicitly? (It's a bit of an annoying song and dance, I know, but: Web.)

Reply:
Changed to `ondevicechanged`